contracts/TokenSale.sol

```diff
@@ -334,6 +318,71 @@ contract TokenSale is FinalizableCrowdsale, Pausable {
         if (investedStarRefund > 0) starToken.transfer(msg.sender, investedStarRefund);
     }
 
+    function verifyTargetRates() internal view {
+        require(
+            targetRates.length == targetRatesTimestamps.length,
+            'Target rates and target rates timestamps should match!'
+        );
+
+        require(targetRates.length > 0, 'Target rates cannot be empty!');
+        require(
+            targetRatesTimestamps[0] == startTime,
+            'First target rate timestamp should match startTime!'
+        );
+
+        for (uint256 i = 0; i < targetRates.length; i++) {
+            if (i > 0) {
+                require(
+                    targetRatesTimestamps[i-1] < targetRatesTimestamps[i],
+                    'Target rates timestamps should be ordered!'
+                );
+            }
+
+            if (i == targetRates.length - 1) {
+               require(
+                    targetRatesTimestamps[i] < endTime,
+                    'All target rate timestamps should be before endTime!'
+                ); 
+            }
+
+            require(targetRates[i] > 0, 'All target rates must above 0!');
+        }
+    }
+
+    /**
+     * @dev Returns current rate and index for rate in targetRates array.
+     * Does not update target rate index, use checkForNewRateAndUpdate() to
+     * update, 
+     */
+    function getCurrentRate() public view returns (uint256, uint256) {
+        for (
+            uint256 i = currentTargetRateIndex + 1;
+            i < targetRatesTimestamps.length;
+            i++
+        ) {
+            if (now < targetRatesTimestamps[i]) {
+                return (targetRates[i - 1], i - 1);
+            }
+        }
+
+        return (
+            targetRates[targetRatesTimestamps.length - 1],
+            targetRatesTimestamps.length - 1
+        );
+    }
+
+    /**
+     * @dev Check for new valid rate and update. Automatically called when
+     * purchasing tokens. 
+     */
+    function checkForNewRateAndUpdate() public {
+        (, uint256 targetRateIndex) = getCurrentRate();
```

Comment:
this `(,` is normal here?

Reply:
Yeah, ignoring the first return value as `getCurrentRate` returns two arguments, but we only need the second one.